File: src/main/java/org/oscarehr/dataMigration/model/appointment/Appointment.java
/**
 * Copyright (c) 2012-2018. CloudPractice Inc. All Rights Reserved.
 * This software is published under the GPL GNU General Public License.
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 *
 * This software was written for
 * CloudPractice Inc.
 * Victoria, British Columbia
 * Canada
 */
package org.oscarehr.dataMigration.model.appointment;

import lombok.Data;
import org.oscarehr.dataMigration.model.AbstractTransientModel;
import org.oscarehr.demographic.model.DemographicModel;
import org.oscarehr.dataMigration.model.provider.Provider;

import java.time.Duration;
import java.time.LocalDateTime;

@Data
public class Appointment extends AbstractTransientModel
{
	private Integer id;

	private Provider provider;

	private LocalDateTime appointmentStartDateTime;
	private LocalDateTime appointmentEndDateTime;

	private String name;
	private DemographicModel demographic;

	private String notes;
	private String reason;
	private String resources;
	private String type;
	private String style;
	private AppointmentStatus status;

	private String location;
	private Site site;

	public Duration getDuration()
	{
		return Duration.between(appointmentStartDateTime, appointmentEndDateTime);
	}

	public long getDurationMin()
	{
		return getDuration().toMinutes() + 1;
	}
}

File: src/main/frontend/src/lib/appointment/model/Appointment.ts
import SimpleProvider from "../../provider/model/SimpleProvider";
import {Moment} from "moment";
import AppointmentStatus from "./AppointmentStatus";

export default class Appointment
{
	private readonly _id: number;

	private _provider: SimpleProvider;
	private _appointmentStartDateTime: Moment;
	private _appointmentEndDateTime: Moment;
	private _name: string;
	private _demographicId: number;

	private _notes: string;
	private _reason: string;
	private _resources: string;
	private _type: string;
	private _style: string;
	private _status: AppointmentStatus;
	private _location: string;
	private _siteId: number;

	constructor(id)
	{
		this._id = id;
	}

	get id(): number
	{
		return this._id;
	}

	get provider(): SimpleProvider
	{
		return this._provider;
	}

	set provider(value: SimpleProvider)
	{
		this._provider = value;
	}

	get appointmentStartDateTime(): Moment
	{
		return this._appointmentStartDateTime;
	}

	set appointmentStartDateTime(value: Moment)
	{
		this._appointmentStartDateTime = value;
	}

	get appointmentEndDateTime(): Moment
	{
		return this._appointmentEndDateTime;
	}

	set appointmentEndDateTime(value: Moment)
	{
		this._appointmentEndDateTime = value;
	}

	get name(): string
	{
		return this._name;
	}

	set name(value: string)
	{
		this._name = value;
	}

	get demographicId(): number
	{
		return this._demographicId;
	}

	set demographicId(value: number)
	{
		this._demographicId = value;
	}

	get notes(): string
	{
		return this._notes;
	}

	set notes(value: string)
	{
		this._notes = value;
	}

	get reason(): string
	{
		return this._reason;
	}

	set reason(value: string)
	{
		this._reason = value;
	}

	get resources(): string
	{
		return this._resources;
	}

	set resources(value: string)
	{
		this._resources = value;
	}

	get type(): string
	{
		return this._type;
	}

	set type(value: string)
	{
		this._type = value;
	}

	get style(): string
	{
		return this._style;
	}

	set style(value: string)
	{
		this._style = value;
	}

	get status(): AppointmentStatus
	{
		return this._status;
	}

	set status(value: AppointmentStatus)
	{
		this._status = value;
	}

	get location(): string
	{
		return this._location;
	}

	set location(value: string)
	{
		this._location = value;
	}

	get siteId(): number
	{
		return this._siteId;
	}

	set siteId(value: number)
	{
		this._siteId = value;
	}
}